app/utils/ipfs.js

```diff
@@ -47,17 +52,47 @@ export async function saveText(text) {
 }
 
 async function saveFileTheGraph(buf) {
-  const saveResult = await getIpfsApiTheGraph().add(buf);
-  // return saveResult.cid.toString();
+  return await getIpfsApiTheGraph().add(buf);
+}
+
+function prepareFile(data) {
+  return [new File([data], 'data')];
+}
+
+async function saveFileWEB3Storage(data) {
+  const token = process.env.WEB3_STORAGE_API_TOKEN;
+
+  if (!token) {
+    return console.error(
+      'A token is needed. You can create one on https://web3.storage',
+    );
+  }
+
+  const storage = new Web3Storage({ token });
+
+  return await storage.put(prepareFile(data));
 }
 
 export async function saveFile(file) {
```

Comment:
That same code has to be changed from `saveText` function. Create a function that is reused in both places.

Reply:
done